### DISCUSSION TITLE
react-firebaseui alternative

### DISCUSSION THREAD

**Discussion by louisnt-prod** [gladly-team/next-firebase-auth#138] [Q&A] [answered]
Hello,

Did anyone find a solution to customize firebase-ui? Or used next-firebase-auth without this component?

`import StyledFirebaseAuth from "react-firebaseui/StyledFirebaseAuth";
`

The workflow of react-firebaseui is super useful to handle email sign-ups but it's super ugly...

Thank you for your help!

**Comment by kmjennison** [comment, marked as answer]
Using `react-firebaseui` is not required for using `next-firebase-auth`. We use it only for the example app. Your app can sign in the Firebase user however you want.

Any feedback on that UI should refer to those projects:
https://github.com/firebase/firebaseui-web-react
https://github.com/firebase/firebaseui-web/

**Reply by louisnt-prod** [reply]
Thank you for the quick reply! I'll try to find how to use of `next-firebase-auth` without `react-firebaseui`.

If I sign in using firebase client, I can still use the hooks and functions from `next-firebase-auth` (useAuthUser for example)?
Should I call the `next-firebase-auth` login endpoint when logging via the firebase client?

**Reply by kmjennison** [reply]
> If I sign in using firebase client, I can still use the hooks and functions from next-firebase-auth (useAuthUser for example)?

Yes.

> Should I call the `next-firebase-auth` login endpoint when logging via the firebase client?

By default, `next-firebase-auth` will handle this by listening to the Firebase JS SDK's events, so you don't have to do it manually.

**Reply by louisnt-prod** [reply]
Amazing! Thanks
